pkg/client/oci/oci.go

```diff
@@ -62,18 +72,75 @@ func New(repo *api.Repo, c cache.Cacher, insecure bool) (*Repo, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	resolver := newDockerResolver(u, repo.GetAuth().GetUsername(), repo.GetAuth().GetPassword(), insecure)
+	// Init entries
+	entries := make(map[string][]string)
 
-	return NewRaw(u, repo.GetAuth().GetUsername(), repo.GetAuth().GetPassword(), c, insecure)
+	// First we try to load the index referenced from config file,
+	// if it does not exist, then we try to guess it directly from repo url
+	ociIndexRef := repo.GetOciIndex()
+	if ociIndexRef == "" {
+		klog.Infof("OCI index reference is empty in config file. Guessing index from repo url")
+		ociIndexRef = fmt.Sprintf("%s%s/index:latest", u.Host, u.Path)
```

Comment:
as follow up in the rename suggestion I made, I'd say:

```suggestion
		chartsIndexRef = fmt.Sprintf("%s%s/index:latest", u.Host, u.Path)
		klog.Infof("Charts index reference has not been set in the config file. Checking default location: %q", chartsIndexRef)
```